cpp/log/cluster_state_controller-inl.h

```diff
@@ -0,0 +1,203 @@
+#ifndef CERT_TRANS_LOG_CLUSTER_STATE_CONTROLLER_INL_H_
+#define CERT_TRANS_LOG_CLUSTER_STATE_CONTROLLER_INL_H_
+
+#include "log/cluster_state_controller.h"
+#include "log/etcd_consistent_store-inl.h"
+
+#include <functional>
+
+namespace cert_trans {
+
+
+template <class Logged>
+ClusterStateController<Logged>::ClusterStateController(
+    util::Executor* executor, ConsistentStore<Logged>* store,
+    const MasterElection* election, const int min_serving_nodes,
+    const double min_serving_fraction)
+    : executor_(CHECK_NOTNULL(executor)),
+      store_(CHECK_NOTNULL(store)),
+      election_(CHECK_NOTNULL(election)),
+      min_serving_nodes_(min_serving_nodes),
+      min_serving_fraction_(min_serving_fraction),
```

Comment:
Maybe add a few sanity CHECKs for ```min_serving_nodes_``` and ```min_serving_fraction_``` in the constructor below? At least 1 node, and, uh... greater than zero, maybe? Greater than or equal to zero?